crates/sui-indexer-alt-jsonrpc/src/api/transactions.rs

```diff
@@ -87,10 +97,11 @@ impl TransactionsApiServer for Transactions {
 
         // Balance changes might not be present because of pruning, in which case we return
         // nothing, even if the changes were requested.
-        let balance_changes = balance_changes
-            .transpose()
-            .map_err(internal_error)?
-            .flatten();
+        let balance_changes = match balance_changes.transpose().map_err(internal_error)? {
```

Comment:
I went back on my choice in #20944 to treat fetching a pruned value as an error that's local to the particular part of the response that was requested. This is because when I tried to apply the same rule to object changes, it didn't seem to work as well: For balance changes, you either get all or nothing (because the data is stored in a single lookaside table), but for object changes, any constituent object's input or output version could be missing which made me lean back towards treating the attempt to fetch a pruned thing as an overall error. I think it would also be reasonable to say that if any constituent object was missing, the overall `object_changes` response field would be `null`, and I'm curious to hear thoughts on what we should go with.